### DISCUSSION TITLE
Release a bugfix version following 4.8.1

### DISCUSSION THREAD

**Discussion by hazendaz** [spotbugs/spotbugs#2691] [General]
Same drill as before, think in about a week here we push a release again.  Would like to have things very stable before the gradle plugin reached 6.0.0 GA.

**Comment by hazendaz** [comment]
@gtoison @JuditKnoll Do we want to wait on the last couple items being worked to make it to master before cutting 4.8.2 or do you think its ready now?  I can do it anytime this week.

**Reply by iloveeclipse** [reply]
> I can do it anytime this week.

We are ready to go now. I would really appreciate if you could push a release this week.

**Comment by iloveeclipse** [comment]
I personally would like to see 4.8.2 ASAP, due the Eclipse 4.30 release that will be there in a week (so anyone who updates could also get fixed spotbugs plugin).

**Comment by gtoison** [comment]
Agreed with ASAP, we fixed a couple of issues

**Comment by JuditKnoll** [comment]
There was new feature added, so it should be 4.9.0, isn't it?
It would be nice if https://github.com/spotbugs/spotbugs/pull/2716 could be merged, but it's not a blocking issue. Let's have the next release ASAP.

**Reply by iloveeclipse** [reply]
> It would be nice if #2716 could be merged

Please rebase your PR on master & it can be merged, the PR is approved already.

> There was new feature added, so it should be 4.9.0, isn't it?

I would argue the bump of the minor segment would mean changed/added API which is not the case here, it was "just" a detector added. So 4.8.2 looks OK for me.

**Reply by JuditKnoll** [reply]
Already merged master into my branch 😄  
You are right, 4.8.2 is okay.

**Reply by iloveeclipse** [reply]
#2716 is merged now.

**Comment by hazendaz** [comment]
Thanks will release tonight.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by perlun** [comment]
4.8.2 is released now, thanks for this. :tada: It doesn't seem to be available on mvnrepository.com though, is publishing it there a manual process? https://mvnrepository.com/artifact/com.github.spotbugs/spotbugs

**Comment by hazendaz** [comment]
Oops, it's staged in sonatype and needs closed out most likely.  I will get to it in a few...

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by hazendaz** [reply]
Its released now, should show available within next 2 hours in the indexes but most likely will start popping pull requests with renovate shortly.

**Reply by JuditKnoll** [reply]
Was this a bug, or should we write something about this in the `RELEASE_PROCEDURE.md`?

**Reply by hazendaz** [reply]
It may be on there, didn't look.  Its not a bug.  It pushes to sonatype staging and closes the release but it does not auto release it.  That could be fixed to just auto release it but it typically gives for a final check before doing so in case something is off.  I had simply logged off last night after pushing the tag up and the job had not finished.  I would have realized it even without anyone noting this when Renovate didn't immediately start sending PRs all over the place.  Typically that will happen within 30 minutes of it being final.

**Reply by JuditKnoll** [reply]
Still 4.8.1 is the latest release on https://mvnrepository.com/artifact/com.github.spotbugs/spotbugs. @hazendaz Can you please look into this?

**Reply by gtoison** [reply]
The new version is on maven: https://repo1.maven.org/maven2/com/github/spotbugs/spotbugs/4.8.2/
It usually takes more time for mvnrepository to pick up the new versions

**Reply by JuditKnoll** [reply]
Great, thanks for the info 😄

**Reply by hazendaz** [reply]
Both central and mvn repository show it now.  And yes it takes some time.  Its supposed to happen within 2 hours but that has gotten worse over last couple of years.  I also believe that night of release sonatype was offline for a while.  I had issues pushing to it and was getting 400 errors on another project and 504 or something like that when trying to log on.  By the time I shut down for the night it was back up but the 400 errors were persisting so I think they were going through some sort of maintenance if i had to guess.

Typically though the moment its out of staging on sonatype, its actually live.  The bots - both renovate and dependabot and probably others will start raising PRs very quickly after.
